Sources/MongoSwift/BSON/Document+Collection.swift

```diff
@@ -4,44 +4,56 @@ import Foundation
 /// This gives guarantees on non-destructive iteration, and offers an indexed
 /// ordering to the key-value pairs in the document.
 extension Document: Collection {
+    /// The index type of a document.
+    public typealias Index = Int
+
     /// Returns the start index of the Document.
-    public var startIndex: Int {
+    public var startIndex: Index {
         return 0
     }
 
     /// Returns the end index of the Document.
-    public var endIndex: Int {
+    public var endIndex: Index {
         return self.count
     }
 
-    private func failIndexCheck(_ i: Int) {
+    private func failIndexCheck(_ i: Index) {
         let invalidIndexMsg = "Index \(i) is invalid"
         guard !self.isEmpty && self.startIndex ... self.endIndex - 1 ~= i else {
             fatalError(invalidIndexMsg)
         }
     }
 
     /// Returns the index after the given index for this Document.
-    public func index(after i: Int) -> Int {
+    public func index(after i: Index) -> Index {
         // Index must be a valid one, meaning it must exist somewhere in self.keys.
         failIndexCheck(i)
         return i + 1
     }
 
     /// Allows access to a `KeyValuePair` from the `Document`, given the position of the desired `KeyValuePair` held
     /// within. This method does not guarantee constant-time (O(1)) access.
-    public subscript(position: Int) -> KeyValuePair {
+    public subscript(position: Index) -> KeyValuePair {
         // TODO: This method _should_ guarantee constant-time O(1) access, and it is possible to make it do so. This
         // criticism also applies to key-based subscripting via `String`.
         // See SWIFT-250.
         failIndexCheck(position)
-        // swiftlint:disable:next force_unwrapping - failIndexCheck precondition ensures non-nil result.
-        return DocumentIterator.subsequence(of: self, startIndex: position, endIndex: position + 1).first!
+        guard let iter = DocumentIterator(forDocument: self) else {
+            fatalError("Failed to initialize an iterator over document \(self)")
+        }
+
+        for pos in 0..<position + 1 {
```

Comment:
nit: maybe clearer to just write this as `for pos in 0...position`?

Reply:
oh yeah duh, silly copy pasta on my part there!